### PR TITLE
New Resource: azurerm_data_lake_store

### PR DESCRIPTION
**Pull request by paktek123**
This PR creates a new resource which can be used for create an Azure Data Lake Store.

### CODE REVIEW

**Comment by paktek123** [comment]
Hi @tombuildsstuff any chance of this being reviewed, would be good to see this in 1.6

File: azurerm/data_source_data_lake_store.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	client := meta.(*ArmClient).dataLakeStoreAccountClient
+	ctx := meta.(*ArmClient).StopContext
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
```
Comment (jeffreyCline) [inline azurerm/data_source_data_lake_store.go]: Can you add a warning message her that it wasn't found?
`log.Printf("[WARN] DataLakeStoreAccount '%s' was not found (resource group '%s').", name, resourcGroup)`

File: azurerm/resource_arm_data_lake_store.go [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["accounts"]
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			fmt.Print("Oh no nothing was found")
```
Comment (jeffreyCline) [inline azurerm/resource_arm_data_lake_store.go]: Can you format the not found message to something like...
`log.Printf("[WARN] DataLakeStore '%s' was not found (resource group '%s')", name, resourcGroup)`

File: azurerm/resource_arm_data_lake_store.go [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+	}
+
+	resourceGroup := id.ResourceGroup
+	name := id.Path["accounts"]
+	future, err := client.Delete(ctx, resourceGroup, name)
+	if err != nil {
+		if response.WasNotFound(future.Response()) {
+			return nil
+		}
+		return fmt.Errorf("Error issuing delete request for Azure Data Lake Store %q (Resource Group %q): %+v", name, resourceGroup, err)
```
Comment (jeffreyCline) [inline azurerm/resource_arm_data_lake_store.go]: Can you update this to be just `return err`?

File: azurerm/resource_arm_data_lake_store.go [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+	}
+
+	err = future.WaitForCompletion(ctx, client.Client)
+	if err != nil {
+		if response.WasNotFound(future.Response()) {
+			return nil
+		}
+		return fmt.Errorf("Error waiting for Azure Data Lake Store %q (Resource Group %q) to be deleted: %+v", name, resourceGroup, err)
```
Comment (jeffreyCline) [inline azurerm/resource_arm_data_lake_store.go]: Same here, just `return err`

File: azurerm/resource_arm_data_lake_store.go [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+	}
+
+	return err
```
Comment (jeffreyCline) [inline azurerm/resource_arm_data_lake_store.go]: Shouldn't this be `return nil`?

**Comment by paktek123** [comment]
Thank you for reviewing. @jeffreyCline , please kindly review again.

File: azurerm/data_source_data_lake_store.go [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+	d.SetId(*resp.ID)
+
+	d.Set("name", name)
+	d.Set("resource_group_name", resourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azureRMNormalizeLocation(*location))
+	}
+
+	if tier := resp.DataLakeStoreAccountProperties; tier != nil {
```
Comment (katbyte) [inline azurerm/data_source_data_lake_store.go]: could we assign this to `properties` not `tier` for clarity?

File: azurerm/data_source_data_lake_store_test.go [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+func TestAccDataSourceAzureRMDataLakeStore_payasyougo(t *testing.T) {
+	dataSourceName := "data.azurerm_data_lake_store.test"
+	rInt := acctest.RandIntRange(1, 999999)
```
Comment (katbyte) [inline azurerm/data_source_data_lake_store_test.go]: Could we use `acctest.RandInt()` like the majority of the other tests?

File: azurerm/data_source_data_lake_store_test.go [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+					testCheckAzureRMDataLakeStoreExists(dataSourceName),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataSourceAzureRMDataLakeStore_monthlycommitment(t *testing.T) {
+	dataSourceName := "data.azurerm_data_lake_store.test"
+	rInt := acctest.RandIntRange(1, 999999)
```
Comment (katbyte) [inline azurerm/data_source_data_lake_store_test.go]: same as above

File: website/docs/r/data_lake_store.html.markdown [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+    location = "westeurope"
+    tier = "Commitment_1TB"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) Specifies the name of the Data Lake Store. Changing this forces a
```
Comment (katbyte) [inline website/docs/r/data_lake_store.html.markdown]: Could we put this all on the same line?

File: azurerm/resource_arm_data_lake_store.go [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+		Create: resourceArmDateLakeStoreCreate,
+		Read:   resourceArmDateLakeStoreRead,
+		Update: resourceArmDateLakeStoreUpdate,
+		Delete: resourceArmDateLakeStoreDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
```
Comment (katbyte) [inline azurerm/resource_arm_data_lake_store.go]: Could we get some validation on this property? the docs say 3-24 so at the very lease we should check that. Ideally we should use regex to also validate the characters.

File: azurerm/data_source_data_lake_store_test.go [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+					resource.TestCheckResourceAttr(dataSourceName, "tier", "Commitment_1TB"),
+					resource.TestCheckResourceAttr(dataSourceName, "tags.%", "1"),
+					resource.TestCheckResourceAttr(dataSourceName, "tags.hello", "world"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataSourceDataLakeStore_payasyougo(rInt int, location string) string {
```
Comment (katbyte) [inline azurerm/data_source_data_lake_store_test.go]: Could we change this to `testAccDataSourceDataLakeStore_basic` as we are checking the basic minimum required tf config?

File: azurerm/data_source_data_lake_store_test.go [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
```
Comment (katbyte) [inline azurerm/data_source_data_lake_store_test.go]: Could we formate the terraform in all the tests?

```
resource "azurerm_resource_group" "test" {
	name     = "acctestRG_%d"
	location = "%s"
}
```

File: azurerm/data_source_data_lake_store_test.go [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+data "azurerm_data_lake_store" "test" {
+	name                = "${azurerm_data_lake_store.test.name}"
+	resource_group_name = "${azurerm_data_lake_store.test.resource_group_name}"
+}
+`, rInt, location, rInt, location)
+}
+
+func testAccDataSourceDataLakeStore_monthlycommitment(rInt int, location string) string {
```
Comment (katbyte) [inline azurerm/data_source_data_lake_store_test.go]: Could we change this to `testAccDataSourceDataLakeStore_tier` as that is what we are checking?

File: azurerm/data_source_data_lake_store_test.go [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+	location := testLocation()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceDataLakeStore_payasyougo(rInt, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDataLakeStoreExists(dataSourceName),
```
Comment (katbyte) [inline azurerm/data_source_data_lake_store_test.go]: Could we also check to make sure that `tier` is set to its default value here?

```
resource.TestCheckResourceAttr(dataSourceName, "tier", "Consumption"),
```

File: azurerm/data_source_data_lake_store_test.go [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+resource "azurerm_resource_group" "test" {
+	name = "acctestRG_%d"
+	location = "%s"
+}
+
+resource "azurerm_data_lake_store" "test" {
+	name = "acctest%d"
+	location = "%s"
+	resource_group_name = "${azurerm_resource_group.test.name}"
+	tags {
```
Comment (katbyte) [inline azurerm/data_source_data_lake_store_test.go]: we should only be checking required fields in the basic test.

File: azurerm/import_arm_data_lake_store_test.go [hunk 16]
```diff
@@ -0,0 +1,54 @@
+package azurerm
```
Comment (katbyte) [inline azurerm/import_arm_data_lake_store_test.go]: We are moving away from a separate import test. Lets remove this file and add the following to all the resource tests:
```
			{
				ResourceName:      resourceName,
				ImportState:       true,
				ImportStateVerify: true,
			},
```

File: azurerm/provider.go [hunk 17]
```diff
@@ -85,6 +85,7 @@ func Provider() terraform.ResourceProvider {
 			"azurerm_client_config":                         dataSourceArmClientConfig(),
 			"azurerm_cosmosdb_account":                      dataSourceArmCosmosDBAccount(),
 			"azurerm_dns_zone":                              dataSourceArmDnsZone(),
+			"azurerm_data_lake_store":                       dataSourceArmDataLakeStoreAccount(),
```
Comment (katbyte) [inline azurerm/provider.go]: minor: alphabetically this should be above `azurerm_dns_zone`

File: azurerm/provider.go [hunk 18]
```diff
@@ -334,6 +336,7 @@ func determineAzureResourceProvidersToRegister(providerList []resources.Provider
 		"Microsoft.DBforPostgreSQL":     {},
 		"Microsoft.Devices":             {},
 		"Microsoft.DocumentDB":          {},
+		"Microsoft.DataLakeStore":       {},
```
Comment (katbyte) [inline azurerm/provider.go]: minor: alphabetically this should be higher up

File: azurerm/resource_arm_data_lake_store.go [hunk 19]
```diff
@@ -0,0 +1,193 @@
...
+				ForceNew: true,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"tier": {
+				Type:             schema.TypeString,
+				ForceNew:         false,
```
Comment (katbyte) [inline azurerm/resource_arm_data_lake_store.go]: we don't need to specify `false` for `ForceNew` so can we remove this line?

File: azurerm/resource_arm_data_lake_store_test.go [hunk 20]
```diff
@@ -0,0 +1,208 @@
...
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAzureRMDataLakeStore_payasyougo(t *testing.T) {
```
Comment (katbyte) [inline azurerm/resource_arm_data_lake_store_test.go]: Could we change this to `TestAccAzureRMDataLakeStore_basic`?

File: azurerm/resource_arm_data_lake_store_test.go [hunk 21]
```diff
@@ -0,0 +1,208 @@
...
+	resourceName := "azurerm_data_lake_store.test"
+	ri := acctest.RandIntRange(1, 999999)
```
Comment (katbyte) [inline azurerm/resource_arm_data_lake_store_test.go]: Could we use `acctest.RandInt()` for the tests here?

File: azurerm/resource_arm_data_lake_store_test.go [hunk 22]
```diff
@@ -0,0 +1,208 @@
...
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDataLakeStoreExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "tier", "Consumption"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMDataLakeStore_monthlycommitment(t *testing.T) {
```
Comment (katbyte) [inline azurerm/resource_arm_data_lake_store_test.go]: Could we change this to `TestAccAzureRMDataLakeStore_tier`?

File: azurerm/resource_arm_data_lake_store_test.go [hunk 23]
```diff
@@ -0,0 +1,208 @@
...
+
+		return fmt.Errorf("Data Lake Store still exists:\n%#v", resp)
+	}
+
+	return nil
+}
+
+func testAccAzureRMDataLakeStore_payasyougo(rInt int, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
```
Comment (katbyte) [inline azurerm/resource_arm_data_lake_store_test.go]: Could we format all these tests?
```
resource "azurerm_resource_group" "test" {
    name     = "acctestRG_%d"
    location = "%s"
}
```

File: website/docs/d/data_lake_store.html.markdown [hunk 24]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+output "data_lake_store_id" {
+  value = "${data.azurerm_data_lake_store.test.id}"
+}
+```
+
+## Argument Reference
+
+* `name` - (Required) The name of the DNS Zone.
```
Comment (katbyte) [inline website/docs/d/data_lake_store.html.markdown]: I think this should be 'The name of the Data Lake Store'

File: website/docs/d/data_lake_store.html.markdown [hunk 25]
```diff
@@ -0,0 +1,36 @@
...
+* `resource_group_name` - (Required) The Name of the Resource Group where the DNS Zone exists.
```
Comment (katbyte) [inline website/docs/d/data_lake_store.html.markdown]: same as above

File: website/docs/r/data_lake_store.html.markdown [hunk 26]
```diff
@@ -0,0 +1,70 @@
+---
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_data_lake_store"
+sidebar_current: "docs-azurerm-resource-data-lake-store"
+description: |-
+  Create a Azure Data Lake Store.
+---
+
+# azurerm\_data\_lake\_store
```
Comment (katbyte) [inline website/docs/r/data_lake_store.html.markdown]: the `\`s are not required here

File: website/docs/r/data_lake_store.html.markdown [hunk 27]
```diff
@@ -0,0 +1,70 @@
+---
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_data_lake_store"
+sidebar_current: "docs-azurerm-resource-data-lake-store"
+description: |-
+  Create a Azure Data Lake Store.
```
Comment (katbyte) [inline website/docs/r/data_lake_store.html.markdown]: Can we change this to 'manage an Azure Data Lake Store"?

File: website/docs/r/data_lake_store.html.markdown [hunk 28]
```diff
@@ -0,0 +1,70 @@
...
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) Specifies the name of the Data Lake Store. Changing this forces a
+    new resource to be created. Has to be between 3 to 24 characters.
+
+* `resource_group_name` - (Required) The name of the resource group in which to
```
Comment (katbyte) [inline website/docs/r/data_lake_store.html.markdown]: same as above

File: website/docs/r/data_lake_store.html.markdown [hunk 29]
```diff
@@ -0,0 +1,70 @@
...
+## Example Usage
+
+```hcl
+# Pay As You Go
+resource "azurerm_resource_group" "test" {
+    name = "test"
+    location = "northeurope"
+}
+
+resource "azurerm_data_lake_store" "payasyougo" {
```
Comment (katbyte) [inline website/docs/r/data_lake_store.html.markdown]: Can we change this to consumption rather the payasyougo to better match the tier name in azure?

File: website/docs/r/data_lake_store.html.markdown [hunk 30]
```diff
@@ -0,0 +1,70 @@
...
+
+# azurerm\_data\_lake\_store
+
+Create a Azure Data Lake Store.
+
+## Example Usage
+
+```hcl
+# Pay As You Go
+resource "azurerm_resource_group" "test" {
```
Comment (katbyte) [inline website/docs/r/data_lake_store.html.markdown]: Can we get the terraform in the example formatted to look nice please?

File: azurerm/resource_arm_data_lake_store_test.go [hunk 31]
```diff
@@ -0,0 +1,208 @@
...
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMDataLakeStoreDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDataLakeStoreExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "tier", "Commitment_1TB"),
+				),
+			},
```
Comment (katbyte) [inline azurerm/resource_arm_data_lake_store_test.go]: Could we get import checks on all these tests?

```
{
				ResourceName:      resourceName,
				ImportState:       true,
				ImportStateVerify: true,
			},
```

File: azurerm/resource_arm_data_lake_store.go [hunk 32]
```diff
@@ -0,0 +1,193 @@
...
+		Location: &location,
+		Tags:     expandTags(tags),
+		CreateDataLakeStoreAccountProperties: &account.CreateDataLakeStoreAccountProperties{
+			NewTier: account.TierType(tier),
+		},
+	}
+
+	future, err := client.Create(ctx, resGroup, name, dateLakeStore)
+	if err != nil {
+		return err
```
Comment (katbyte) [inline azurerm/resource_arm_data_lake_store.go]: Could we wrap these errors? like how you are in the update
```
future, err := client.Update(ctx, resourceGroup, name, props)
	if err != nil {
		return fmt.Errorf("Error issuing update request for Data Lake Store %q (Resource Group %q): %+v", name, resourceGroup, err)
	}
```

File: azurerm/data_source_data_lake_store.go (hunk 6 above)
Comment (paktek123) [reply azurerm/data_source_data_lake_store.go]: fixed in next commit

File: azurerm/data_source_data_lake_store_test.go (hunk 7 above)
Comment (paktek123) [reply azurerm/data_source_data_lake_store_test.go]: there is a limitation on the name for an ADLS account which has to be between 3  to 24 characters. I wanted to restrict the length of the name to within that limit. RandInt() generates a larger string if I am not mistaken.

File: azurerm/data_source_data_lake_store_test.go (hunk 8 above)
Comment (paktek123) [reply azurerm/data_source_data_lake_store_test.go]: there is a limitation on the name for an ADLS account which has to be between 3  to 24 characters. I wanted to restrict the length of the name to within that limit. RandInt() generates a larger string if I am not mistaken.

File: website/docs/r/data_lake_store.html.markdown (hunk 9 above)
Comment (paktek123) [reply website/docs/r/data_lake_store.html.markdown]: fixed in next commit

File: azurerm/resource_arm_data_lake_store.go (hunk 10 above)
Comment (paktek123) [reply azurerm/resource_arm_data_lake_store.go]: done in next commit

File: azurerm/data_source_data_lake_store_test.go (hunk 11 above)
Comment (paktek123) [reply azurerm/data_source_data_lake_store_test.go]: done, renamed to basic in next commit

File: azurerm/data_source_data_lake_store_test.go (hunk 12 above)
Comment (paktek123) [reply azurerm/data_source_data_lake_store_test.go]: moved to 2 spaces (I think thats what I think that means)

File: azurerm/data_source_data_lake_store_test.go (hunk 13 above)
Comment (paktek123) [reply azurerm/data_source_data_lake_store_test.go]: done in next commit

File: azurerm/data_source_data_lake_store_test.go (hunk 14 above)
Comment (paktek123) [reply azurerm/data_source_data_lake_store_test.go]: done in next commit

File: azurerm/data_source_data_lake_store_test.go (hunk 15 above)
Comment (paktek123) [reply azurerm/data_source_data_lake_store_test.go]: yup makes sensem, removed tags

File: azurerm/import_arm_data_lake_store_test.go (hunk 16 above)
Comment (paktek123) [reply azurerm/import_arm_data_lake_store_test.go]: done, in next commit

File: azurerm/provider.go (hunk 17 above)
Comment (paktek123) [reply azurerm/provider.go]: moved, in next commit

File: azurerm/provider.go (hunk 18 above)
Comment (paktek123) [reply azurerm/provider.go]: moved in next commit

File: azurerm/resource_arm_data_lake_store.go (hunk 19 above)
Comment (paktek123) [reply azurerm/resource_arm_data_lake_store.go]: yup not needed

File: azurerm/resource_arm_data_lake_store_test.go (hunk 20 above)
Comment (paktek123) [reply azurerm/resource_arm_data_lake_store_test.go]: changed

File: azurerm/resource_arm_data_lake_store_test.go (hunk 21 above)
Comment (paktek123) [reply azurerm/resource_arm_data_lake_store_test.go]: there is a limitation on the name for an ADLS account which has to be between 3 to 24 characters. I wanted to restrict the length of the name to within that limit. RandInt() generates a larger string if I am not mistaken.

File: azurerm/resource_arm_data_lake_store_test.go (hunk 22 above)
Comment (paktek123) [reply azurerm/resource_arm_data_lake_store_test.go]: done

File: azurerm/resource_arm_data_lake_store_test.go (hunk 23 above)
Comment (paktek123) [reply azurerm/resource_arm_data_lake_store_test.go]: done, moved to 2 spaces

File: website/docs/d/data_lake_store.html.markdown (hunk 24 above)
Comment (paktek123) [reply website/docs/d/data_lake_store.html.markdown]: yes :), fixed

File: website/docs/d/data_lake_store.html.markdown (hunk 25 above)
Comment (paktek123) [reply website/docs/d/data_lake_store.html.markdown]: fixed, in next commit

File: azurerm/resource_arm_data_lake_store.go (hunk 32 above)
Comment (paktek123) [reply azurerm/resource_arm_data_lake_store.go]: done for Create

File: azurerm/resource_arm_data_lake_store_test.go (hunk 31 above)
Comment (paktek123) [reply azurerm/resource_arm_data_lake_store_test.go]: and removed import file

**Comment by paktek123** [comment]
Thank you for reviewing @katbyte. Please kindly review again.

File: azurerm/data_source_data_lake_store.go [hunk 33]
```diff
@@ -0,0 +1,68 @@
...
+				Required: true,
+			},
+
+			"resource_group_name": resourceGroupNameForDataSourceSchema(),
+
+			"location": locationForDataSourceSchema(),
+
+			"tier": {
+				Type:     schema.TypeString,
+				Optional: true,
```
Comment (katbyte) [inline azurerm/data_source_data_lake_store.go]: I believe this should be `computed` with no default?

File: azurerm/resource_arm_data_lake_store.go [hunk 34]
```diff
@@ -0,0 +1,204 @@
...
+	}
+
+	future, err := client.Create(ctx, resourceGroup, name, dateLakeStore)
+	if err != nil {
+		return fmt.Errorf("Error issuing create request for Data Lake Store %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	err = future.WaitForCompletion(ctx, client.Client)
+	if err != nil {
+		return err
```
Comment (katbyte) [inline azurerm/resource_arm_data_lake_store.go]: Could we wrap this error like above?

File: azurerm/resource_arm_data_lake_store.go [hunk 35]
```diff
@@ -0,0 +1,204 @@
...
+	}
+
+	read, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		return err
```
Comment (katbyte) [inline azurerm/resource_arm_data_lake_store.go]: Could we wrap this error like above?

File: azurerm/resource_arm_data_lake_store.go [hunk 36]
```diff
@@ -0,0 +1,204 @@
...
+		if response.WasNotFound(future.Response()) {
+			return nil
+		}
+		return err
+	}
+
+	return nil
+}
+
+func validateArmDataLakeName(v interface{}, k string) (ws []string, es []error) {
```
Comment (katbyte) [inline azurerm/resource_arm_data_lake_store.go]: Very minor: as it doesn't look like this is used anywhere else you could just use the built in validation helpers:
```go
                ValidateFunc: validation.StringMatch(
					regexp.MustCompile(`\A([a-z0-9]{3,24})\z`),
					"Name can only consist of lowercase letters and numbers, and must be between 3 and 24 characters long",
				),
```
But I'll merge as is

File: azurerm/resource_arm_data_lake_store.go [hunk 37]
```diff
@@ -0,0 +1,204 @@
...
+	}
+
+	resourceGroup := id.ResourceGroup
+	name := id.Path["accounts"]
+	future, err := client.Delete(ctx, resourceGroup, name)
+	if err != nil {
+		if response.WasNotFound(future.Response()) {
+			return nil
+		}
+		return err
```
Comment (katbyte) [inline azurerm/resource_arm_data_lake_store.go]: Could we wrap this error too?

File: azurerm/resource_arm_data_lake_store.go [hunk 38]
```diff
@@ -0,0 +1,204 @@
...
+	}
+
+	err = future.WaitForCompletion(ctx, client.Client)
+	if err != nil {
+		if response.WasNotFound(future.Response()) {
+			return nil
+		}
+		return err
```
Comment (katbyte) [inline azurerm/resource_arm_data_lake_store.go]: Could we wrap this error too?

File: azurerm/data_source_data_lake_store.go (hunk 33 above)
Comment (paktek123) [reply azurerm/data_source_data_lake_store.go]: yup has to be computed

File: azurerm/resource_arm_data_lake_store.go (hunk 34 above)
Comment (paktek123) [reply azurerm/resource_arm_data_lake_store.go]: wrapped in next commit

File: azurerm/resource_arm_data_lake_store.go (hunk 35 above)
Comment (paktek123) [reply azurerm/resource_arm_data_lake_store.go]: wrapped in next commit

File: azurerm/resource_arm_data_lake_store.go (hunk 36 above)
Comment (paktek123) [reply azurerm/resource_arm_data_lake_store.go]: makes sense, no need for a func that doesnt get used elsewhere

File: azurerm/resource_arm_data_lake_store.go (hunk 37 above)
Comment (paktek123) [reply azurerm/resource_arm_data_lake_store.go]: wrapped in next commit

File: azurerm/resource_arm_data_lake_store.go (hunk 38 above)
Comment (paktek123) [reply azurerm/resource_arm_data_lake_store.go]: wrapped in next commit

**Comment by paktek123** [comment]
Thank you for reviewing @katbyte . Please kindly review again.

**Pull request merged**